### DISCUSSION TITLE
Feature: Only show translated languages in Mapeo's settings screen

### DISCUSSION THREAD

**Discussion by gmaclennan** [digidem/mapeo-mobile#801] [Ideas]
**Current Behavior:**
Currently, the settings screen in Mapeo shows a list of all languages that are [activated for translation in CrowdIn](https://translate.mapeo.app). However, some of these languages might not be translated, or only partially translated, so that when the user changes language they do not get the expected result, and Mapeo continues to appear in English.

**Desired Behavior:**
Only show languages that have been translated as language options in the Mapeo settings screen.

**Question:**
How complete should a translation be before we show it as an option in Mapeo? I think it's clear that if 0% is translated we should not show it, but if only 1 or 2 key strings are translated, that could still be useful for a user. However, changing language but then seeing everything in English still can be a frustrating user experience.

My suggestion would be to set a cut-off of 50%, e.g. the user is not shown a language as an option in Mapeo until 50% of the strings are translated. We _could_ choose this based on a list of certain "key" strings, e.g. the strings that show on the main pages, but that would be hard to define and maintain.

Interested in thoughts from @arky @rudokemper @jencastrodoesstuff @aliya-ryan (this is non-urgent).

**Comment by arky** [comment]
50% should work. In future, it should be possible to separate main screens/stories into individual files and give them higher priority. https://support.crowdin.com/files-management/#prioritizing-files 

I would recommend an active shipping language system (as followed in other open source projects). An active shipping language is defined as the one with at least one active translator or backed by community of stackholders who are actively use the tool in that language.  

If a active shipping language misses translation deadline due to attrition rate for couple of development release cycles then they would be removed from the release train.
